sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDeviceManager.scala

```diff
@@ -217,6 +224,11 @@ object GpuDeviceManager extends Logging {
     if (!Rmm.isInitialized) {
       val conf = rapidsConf.getOrElse(new RapidsConf(SparkEnv.get.conf))
       val info = Cuda.memGetInfo()
+
+      // We need to reserve more memory when RAPIDS shuffle is enabled and we are using the CUDA
```

Comment:
So the `GpuShuffleEnv.init(conf)`  doesn't end up allocating the bounce buffers. We are accounting for the reserve up above, so I think the init call should just return to what it was before.

What do you think?

Reply:
We are calling `GpuShuffleEnv.shouldUseRapidsShuffle(conf)` before initializing RMM, which requires `GpuShuffleEnv` to be initialized.

Reply:
That's absolutely right.